src/misc/types.d.ts

```diff
@@ -0,0 +1,12 @@
+export type WithOkAndErrorMsg<T> = {
```

Comment:
Have you considered using a tagged union instead?

```
{
	success: true,
	inner: T
} | {
	success: false,
	errorMsg: string
}
```

Reply:
Yeah, that's much better. I will implement it. Thanks.